cmd/frontend/auth/redirect.go

```diff
@@ -23,3 +24,10 @@ func SafeRedirectURL(urlStr string) string {
 	u = &url.URL{Path: u.Path, RawQuery: u.RawQuery}
 	return u.String()
 }
+
+// Redirects to sign in page to display error messages after third-party auth errors.
+//
+// 🚨 SECURITY: The `message` must not contain any confidential information.
+func ProviderErrorRedirect(w http.ResponseWriter, r *http.Request, message string) {
+	http.Redirect(w, r, "/sign-in?auth_error="+message, http.StatusFound)
```

Comment:
Just noticed the `message` might contain spaces, have you had chance to verify it works in major browsers? Otherwise, we might need to escape spaces ourselves here:

```suggestion
	http.Redirect(w, r, "/sign-in?auth_error="+url.QueryEscape(message), http.StatusFound)
```